src/sagemaker_tensorflow_container/training.py

```diff
@@ -13,16 +13,127 @@
 
 from __future__ import absolute_import
 
+import json
 import logging
 
 import sagemaker_containers.beta.framework as framework
 
+
 logger = logging.getLogger(__name__)
 
 
+SAGEMAKER_PARAMETER_SERVER_NUM = 'sagemaker_parameter_server_num'
+
+
+def _is_host_master(hosts, current_host):
+    return current_host == hosts[0]
+
+
+def _build_tf_config(hosts, current_host, ps_num=0, ps_task=False):
+    """
+    Builds a dictionary containing cluster information based on number of hosts and number of
+    parameter servers.
+
+    Args:
+        hosts (list[str]): List of host name in the cluster
+        current_host (str): Current host name
+        ps_num (int): Number of parameter servers
+        ps_task (bool): Set to True if this config is built for a ps server process
+
+    Returns:
+        dict[str: dict]: A dictionary describing the cluster setup for distributed training.
+        For more information regarding TF_CONFIG:
+        https://cloud.google.com/ml-engine/docs/tensorflow/distributed-training-details
+
+    """
```

Comment:
nit: don't need a newline before the final """

Reply:
sure